src/main/java/edu/ucla/library/lambda/kakadu/converter/Kakadu.java

```diff
@@ -55,25 +57,35 @@
      *
      * @param aID An ID for the image to be converted
      * @param aTIFF A TIFF image to be converted
-     * @param aConversion A type of conversion (e.g. lossy, lossless)
+     * @param aCompressionRate A type of conversion (e.g. lossy, lossless)
      * @return The JP2 file
      * @throws IOException If there is trouble reading the source image or writing the JP2
      * @throws InterruptedException If the process gets interrupted
+     * @throws IllegalArgumentException If the compression rate isn't dash or a float
      */
-    public File convert(final String aID, final File aTIFF, final Conversion aConversion) throws IOException,
-            InterruptedException {
+    public File convert(final String aID, final File aTIFF, final String aCompressionRate) throws IOException,
+            InterruptedException, IllegalArgumentException {
         final String id = aID.contains("/") ? URLEncoder.encode(aID, StandardCharsets.UTF_8.toString()) : aID;
         final File jp2 = new File(TMP_DIR, id + JP2_EXT);
         final List<String> command = new ArrayList<>();
-        final String conversion = aConversion.name();
 
         command.addAll(Arrays.asList(KAKADU_COMMAND, "-i", getPath(aTIFF), "-o", getPath(jp2)));
         command.addAll(BASE_OPTIONS);
 
-        if (conversion.equals(Conversion.LOSSLESS.name())) {
+        if (aCompressionRate.equals(LOSSLESS_RATE)) {
             command.addAll(LOSSLESS_OPTIONS);
-        } else if (conversion.equals(Conversion.LOSSY.name())) {
-            command.addAll(LOSSY_OPTION);
+        } else {
+            final List<String> options = new ArrayList<>(LOSSY_OPTIONS);
+
+            // Confirm our lossy compression rate value is a float
+            try {
+                Float.parseFloat(aCompressionRate);
```

Comment:
Should we also make sure that the compression rate is greater than zero? My reading of the [usage docs](https://gist.github.com/ahankinson/4945722#file-gistfile1-txt-L173) indicates that it can't be negative or zero.

Reply:
Yes, good idea. I'll add it.